src/lib/Libecl/ecl_verify_values.c

```diff
@@ -1106,12 +1106,13 @@ verify_value_zero_or_positive(int batch_request, int parent_object, int cmd,
 	struct attropl *pattr, char **err_msg)
 {
 	long l;
+	char *end = NULL;
 
 	if ((pattr->value == NULL) || (pattr->value[0] == '\0'))
 		return PBSE_BADATVAL;
 
-	l = atol(pattr->value);
-	if (l < 0)
+	l = strtol(pattr->value, &end, 10);
+	if ((errno == ERANGE) || (errno == EINVAL) || (l < 0))
```

Comment:
Please initialize errno to zero before the call to strtol() to ensure the errno is not stale. When you do this, you can (optionally) remove the specific tests against ERANGE and EINVAL and just check if errno != 0.

It would be good to rename the variable "l" to something like "lval". Othewise, it's easy to mistake for the integer 1.